Stats/src/main/java/io/deephaven/stats/ObjectAllocationCollector.java

```diff
@@ -12,13 +12,14 @@
 import io.deephaven.base.stats.Stats;
 import io.deephaven.base.stats.Value;
 
+import java.util.Objects;
+
 /**
- * Use the allocation instrumenter from http://code.google.com/p/java-allocation-instrumenter/ to produce stats
+ * Use the allocation instrumenter from https://github.com/google/allocation-instrumenter to produce stats
  *
  * To use make sure you set
  *
- * -j -Dallocation.stats.enabled=true -j -Xbootclasspath/a:java_lib/allocation-2.0.jar -j
- * -javaagent:java_lib/allocation-2.0.jar (maybe even use -j -noverify)
+ * -Dallocation.stats.enabled=true -javaagent:/opt/deephaven/server/lib/java-allocation-instrumenter-3.3.0.jar
```

Comment:
This comment implies we are adding the jar to our classpath - but since it's compileOnly, we *aren't*; yeah?

Reply:
Making this compileOnly makes the feature opt-in, and we want it off by default - without the javaagent set, it doesnt matter if you turn enable `allocation.stats.enabled` or not, nothing will be recorded).

The act of passing javaagent both a) adds that (shaded) jar to the classpath and b) enables the jar's premain to run and instrument the real classpath. As a side effect of b), we will see the jar from our own classpath.

The path in that documentation string is just a hypothetical, but we don't ship the jar (or its ... incorrect dependency tree) at all - I have a comment on #2120 that describes how to actually enable this feature.